test/integration/roles/test_s3/tasks/main.yml

```diff
@@ -2,10 +2,10 @@
 # tasks file for test_s3
 # ============================================================
 - name: generate random name for the bucket name
-  shell: echo ansible_test_$RANDOM
+  shell: bash -c 'echo ansible_test_$RANDOM'
```

Comment:
@s-hertel - I've committed a fairly serious fix to this now.  Please look it over then I think this can be merged.